test/test_linearizer.py

```diff
@@ -171,18 +171,13 @@ def test_range_outer_op_after_phi_nested_range(self):
     end = max(i for i,u in enumerate(lin.uops) if u.uop is UOps.ENDRANGE)
     assert lin.uops[end+1].uop is UOps.ALU
 
-  @unittest.expectedFailure
+  @unittest.skip
+  @unittest.skipUnless(Device[Device.DEFAULT].renderer.has_local, "test requires locals")
   def test_early_end_local(self):
```

Comment:
this tests that `ENDIF` uops are placed properly when there are multiple local reductions
ex it's looking to make sure we get a kernel that looks like
```
for () { ... }
if () { for() { ... } ... )
for () { ... }
if () { for() { ... } ... )
```
instead of:
```
for () { ... }
if () { 
  for() { ... } ... )
  for () { ... }
  if () { for() { ... } ... )
}